test/specs/asyncapi/canHandle.spec.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const $RefParser = require("../../..");
+
+const { expect } = require("chai");
+const { readFileSync } = require("fs");
+const { join } = require("path");
+const AJV = require("ajv");
+
+describe("can process ayncapi schema so that AJV does not choke", () => {
+
+  it("through dereferencing, ignoring circular references", async () => {
+    const [schema, doc] = buildFixtures();
+
+    const derefNoCircular = await $RefParser.dereference(
+      schema,
+      { dereference: { circular: "ignore" }},
+    );
+
+    validate(derefNoCircular, doc);
+  });
+
+  it("through full dereferencing", async () => {
+    const [schema, doc] = buildFixtures();
+
+    const deref = await $RefParser.dereference(
+      schema,
+    );
+
+    validate(deref, doc);
+  });
+
+  it("through bundling", async () => {
+    const [schema, doc] = buildFixtures();
+
+    const bundled = await $RefParser.bundle(
+      schema,
+    );
+
+    validate(bundled, doc);
+  });
+
+  function validate (schema, doc) {
+    const ajv = new AJV({
+      allErrors: true,
+      schemaId: "auto",
```

Comment:
I think this one should be:

```suggestion
      schemaId: "id",
```

Reply:
@fmvilas Yep. That did it. Switching to "id" did it and solved the bundling issue. Thanks for the hint!